src/containers/TransactionPage/TransactionPage.stateDataRescueBooking.js

```diff
@@ -0,0 +1,97 @@
+import {
+  TX_TRANSITION_ACTOR_CUSTOMER as CUSTOMER,
+  TX_TRANSITION_ACTOR_PROVIDER as PROVIDER,
+  CONDITIONAL_RESOLVER_WILDCARD,
+  ConditionalResolver,
+} from '../../transactions/transaction';
+
+/**
+ * Get state data against booking process for TransactionPage's UI.
+ * I.e. info about showing action buttons, current state etc.
+ *
+ * @param {*} txInfo detials about transaction
+ * @param {*} processInfo  details about process
+ */
+export const getStateDataForRescueBookingProcess = (txInfo, processInfo) => {
+  const { transaction, transactionRole, nextTransitions } = txInfo;
+  const isProviderBanned = transaction?.provider?.attributes?.banned;
+  const isCustomerBanned = transaction?.provider?.attributes?.banned;
+  const _ = CONDITIONAL_RESOLVER_WILDCARD;
+
+  const {
+    processName,
+    processState,
+    states,
+    transitions,
+    isCustomer,
+    actionButtonProps,
+    leaveReviewProps,
+  } = processInfo;
+
+  return new ConditionalResolver([processState, transactionRole])
+    .cond([states.INQUIRY, CUSTOMER], () => {
+      const transitionNames = Array.isArray(nextTransitions)
+        ? nextTransitions.map(t => t.attributes.name)
+        : [];
+      const requestAfterInquiry = transitions.REQUEST_AFTER_INQUIRY;
+      const hasCorrectNextTransition = transitionNames.includes(requestAfterInquiry);
+      const showOrderPanel = !isProviderBanned && hasCorrectNextTransition;
+      return { processName, processState, showOrderPanel };
+    })
+    .cond([states.INQUIRY, PROVIDER], () => {
+      return { processName, processState, showDetailCardHeadings: true };
+    })
+    .cond([states.BOOKING_REQUEST_SENT, CUSTOMER], () => {
+      return { processName, processState, showDetailCardHeadings: true, showExtraInfo: true };
+    })
+    .cond([states.BOOKING_REQUEST_SENT, PROVIDER], () => {
+      const primary = isCustomerBanned ? null : actionButtonProps(transitions.ACCEPT, PROVIDER);
+      const secondary = isCustomerBanned ? null : actionButtonProps(transitions.PROVIDER_DECLINE, PROVIDER);
```

Comment:
compare null vs undefined

Reply:
Check this

Reply:
In this situation, null and undefined will give the same result. It will hide the buttons when the customer is banned. But null is different from undefined. From my perspective, null is the value declared but does not point to the specific value, and it is the object null. On the other hand, undefined is nothing, and typeof undefined will give undefined.